### ISSUE TITLE
Is there a way to train with agnostic-nms?

### ISSUE THREAD

**Issue by sarmientoj24** [ultralytics/yolov5#8825]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I would want to trade with agnostic-nms as settings

**Comment by glenn-jocher** [comment]
@sarmientoj24 training doesn't use NMS

**Comment by sarmientoj24** [comment]
how does it evaluate stuff for deciding how to backpropagate or adjust the model's weights?

**Comment by sarmientoj24** [comment]
also, is there a way to prohibit the model (or train it) in such a way that a class can only appear minimum of 0 and maximum of 1 instance in an image?

**Comment by glenn-jocher** [comment]
@​sarmientoj24 The training evaluates loss using backpropagation to adjust the model's weights. Regarding your second question, you can enforce a minimum and maximum number of instances per class by adjusting the dataset and using techniques like data augmentation and custom loss functions. Check out the Ultralytics Docs for more details.
